metricbeat/module/kafka/partition/partition.go

```diff
@@ -69,7 +71,8 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		BaseMetricSet: base,
 		broker:        broker,
 		cfg:           cfg,
-		id:            noID,
+		id:            config.ID,
```

Comment:
Do we really need both config options? Implementation wise for matching id is probably much easier. And the id of a broker should never change over the lifetime of the broker I assume?